vars/dgossExecute.groovy

```diff
@@ -0,0 +1,135 @@
+import com.sap.piper.DockerUtils
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = [
+    /** The name of the docker image to dgoss. */
+    'dockerImageName',
+    /** Defines the registry url where the image should be located*/
+    'dockerRegistryUrl',
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+
+    /** The tag of the image to dgoss. */
+    'dockerImageTag',
+    /**
+     * gossFile The oath to the goss file to use. Default value is 'goss.yaml'.
+     */
+    'gossFile'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * This step execute goss validation agains your container
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        final script = checkScript(this, parameters) ?: this
+        def utils = parameters.juStabUtils ?: new Utils()
+        // handle deprecated parameters
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        DockerUtils dockerUtils = new DockerUtils(script)
+        ConfigurationHelper.newInstance(this, config)
+            .withMandatoryProperty('dockerImageName')
+            .withMandatoryProperty('dockerImageTag')
+            .withMandatoryProperty('gossFile')
+
+        def dockerImageNameAndTag = "${config.dockerImageName}:${config.dockerImageTag}"
+
+
+        if (config.dockerRegistryUrl) {
+            dockerImageNameAndTag = config.dockerRegistryUrl + "/" + dockerImageNameAndTag
+        }
+
+        if (dockerUtils.onKubernetes()){
+            runOnK8S(config, dockerImageNameAndTag)
+
+        }else if (dockerUtils.withDockerDaemon()){
+            runOnNode(dockerImageNameAndTag)
+        }else{
+            error "[${STEP_NAME}] No Docker daemon available, dgoss require a runnig docker daemon"
+        }
+    }
+}
+
+def runOnNode(dockerImageNameAndTag){
```

Comment:
What is this method returning? Please use "void" if you didn't intend for it to return anything.